data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessorITWithAcks.java

```diff
@@ -292,14 +296,14 @@ public void testRateLimiterDropAggregation() throws Exception {
         when(aggregateProcessorConfig.getGroupDuration()).thenReturn(Duration.ofSeconds(GROUP_DURATION_FOR_ONLY_SINGLE_CONCLUDE));
         final Processor processor = new AggregateProcessor(aggregateProcessorConfig, pluginMetrics, pluginFactory, expressionEvaluator);
         processors = List.of(processor);
-        when(pipeline.getProcessors()).thenReturn(processors);
+        when(processorProvider.getProcessors()).thenReturn(processors);
         final FutureHelperResult<Void> futureHelperResult = mock(FutureHelperResult.class);
         when(futureHelperResult.getFailedReasons()).thenReturn(Collections.emptyList());
         try (final MockedStatic<FutureHelper> futureHelperMockedStatic = mockStatic(FutureHelper.class)) {
             futureHelperMockedStatic.when(() -> FutureHelper.awaitFuturesIndefinitely(sinkFutures))
                     .thenReturn(futureHelperResult);
 
-            final ProcessWorker processWorker = new ProcessWorker(buffer, processors, pipeline);
+            final ProcessWorker processWorker = new ProcessWorker(buffer, pipeline);
```

Comment:
There is something wrong with these tests if they are so tightly coupled to our internal processing. This is probably too big to fix in this PR, but something to point out.